packages/iov-bns/src/bnsconnection.ts

```diff
@@ -784,10 +787,22 @@ export class BnsConnection implements AtomicSwapConnection {
     return nfts;
   }
 
-  public async getAccountNft(query: AccountsByNameQuery): Promise<readonly AccountNft[]> {
-    const results = (await this.query("/accounts", toUtf8(query.name))).results;
+  public async getAccounts(query: BnsAccountsQuery): Promise<readonly AccountNft[]> {
+    let keyPrefix = "account:";
+    let results: readonly Result[];
+    if (isBnsAccountByNameQuery(query)) {
+      results = (await this.query("/accounts", toUtf8(query.name))).results;
+    } else if (isBnsAccountsByOwnerQuery(query)) {
+      keyPrefix = "";
+      const rawAddress = decodeBnsAddress(query.owner).data;
+      results = (await this.query("/accounts/owner", rawAddress)).results;
+    } else if (isBnsAccountsByDomainQuery(query)) {
+      results = (await this.query("/accounts/domain", toUtf8(query.domain))).results;
+    } else {
+      throw new Error("Unsupported query");
+    }
 
-    const parser = createParser(codecImpl.account.Account, "account:");
+    const parser = createParser(codecImpl.account.Account, keyPrefix);
```

Comment:
Same issue as https://github.com/iov-one/iov-core/pull/1382/files#r377560991. But key contains full account name with domain instead of id.

Reply:
https://github.com/iov-one/iov-core/pull/1382/files#r378156586